src/dataflow/components/dataflow-program.tsx

```diff
@@ -195,11 +198,23 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   }
 
   private heartBeat = () => {
+    let processNeeded = false;
     this.programEditor.nodes.forEach((n: Node) => {
+      if (n.name === "Generator") {
+        processNeeded = true;
+        this.updateGeneratorNode(n);
+      }
       if (n.data.hasOwnProperty("nodeValue")) {
         this.updateNodeRecentValues(n);
       }
     });
+    if (processNeeded) {
+      (async () => {
+        await this.programEngine.abort();
+        await this.programEngine.process(this.programEditor.toJSON());
```

Comment:
A comment about why this `abort()`/`process()` sequence is required would be helpful.

Reply:
Added.

Reply:
I'm not seeing a new comment in the code.

Reply:
Should be there now.